tests/integration/security/cacert_rotation/main_test.go

```diff
@@ -0,0 +1,192 @@
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/util/protomarshal"
+	"istio.io/istio/tests/integration/security/util/cert"
+	"istio.io/istio/tests/integration/security/util/reachability"
+)
+
+var apps deployment.SingleNamespaceView
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		Label(label.CustomSetup).
+		Setup(istio.Setup(nil, setupConfig, cert.CreateCASecret)).
+		Setup(deployment.SetupSingleNamespace(&apps, deployment.Config{})).
+		Setup(func(ctx resource.Context) error {
+			return reachability.CreateCustomInstances(&apps)
+		}).
+		Run()
+}
+
+func setupConfig(_ resource.Context, cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	cfgYaml := `
+values:
+  pilot:
+    env:
+      ISTIO_MULTIROOT_MESH: true
+  meshConfig:
+    defaultConfig:
+      proxyMetadata:
+        PROXY_CONFIG_XDS_AGENT: "true"
+`
+	cfg.ControlPlaneValues = cfgYaml
+}
+
+func TestReachability(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.peer.cacert-rotation").
+		Run(func(t framework.TestContext) {
+			istioCfg := istio.DefaultConfigOrFail(t, t)
+			istioCtl := istioctl.NewOrFail(t, t, istioctl.Config{})
+			namespace.ClaimOrFail(t, t, istioCfg.SystemNamespace)
+
+			from := apps.EchoNamespace.A
+			to := apps.EchoNamespace.B
+			fromAndTo := from.Append(to)
+
+			lastUpdateTime, err := getWorkloadCertLastUpdateTime(t, from[0], istioCtl)
+			if err != nil {
+				t.Errorf("failed to get workload cert last update time: %v", err)
+			}
+
+			// Verify traffic works between a and b
+			echotest.New(t, fromAndTo).
+				WithDefaultFilters(1, 1).
+				FromMatch(match.ServiceName(from.NamespacedName())).
+				ToMatch(match.ServiceName(to.NamespacedName())).
+				Run(func(t framework.TestContext, from echo.Instance, to echo.Target) {
+					// Verify mTLS works between a and b
+					opts := echo.CallOptions{
+						To: to,
+						Port: echo.Port{
+							Name: "http",
+						},
+					}
+					opts.Check = check.And(check.OK(), check.ReachedTargetClusters(t))
+
+					from.CallOrFail(t, opts)
+				})
+
+			// Rotate CA cert
+			if err := cert.CreateCustomCASecret(t,
+				"ca-cert.pem", "ca-key.pem",
+				"cert-chain.pem", "root-cert-combined.pem"); err != nil {
+				t.Errorf("failed to update combined CA secret: %v", err)
+			}
+
+			// Wait for workload cert to be updated
+			retry.UntilOrFail(t, func() bool {
+				updateTime, err := getWorkloadCertLastUpdateTime(t, from[0], istioCtl)
+				if err != nil {
+					t.Logf("failed to get workload cert last update time: %v", err)
+					return false
+				}
+
+				// retry when workload cert is not updated
+				if updateTime.After(lastUpdateTime) {
+					t.Logf("workload cert is updated, last update time: %v", updateTime)
+					return true
+				}
+
+				return false
+			}, retry.Timeout(time.Minute), retry.Delay(1*time.Second))
+
+			// Verify traffic works between a and b after cert rotation
```

Comment:
Maybe it is feasible to startup a go thread running calling during the waiting, just not sure how to control the calling numbers